nilearn/input_data/base_masker.py

```diff
@@ -38,6 +40,40 @@ def filter_and_mask(imgs, mask_img_,
 
     mask_img_ = _utils.check_niimg(mask_img_, ensure_3d=True)
     imgs = _utils.check_niimgs(imgs, accept_3d=True)
+    if mask_time is None:
+        imgs = _utils.check_niimgs(imgs, accept_3d=True)
+    else:
+        # check type and value of mask_time
+        if (not isinstance(mask_time, list) or
+            len(mask_time) == 0 or
+            not isinstance(mask_time[0], bool)):
```

Comment:
Agreed with Loic. I wouldn't even call 'asarray' on it, so that it is
possible to pass Slice objects.

And do we want to call it 'mask_time' or 'time_mask' :)